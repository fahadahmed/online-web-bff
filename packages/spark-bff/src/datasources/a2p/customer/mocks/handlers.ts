import { getAdminOverviewMock } from './getAdminOverview';
import { getCustomerOverviewMock } from './getCustomerOverview';
import { getContentProvidersMock } from './getContentProviders';
import { getServiceRequestsMock } from './getServiceRequests';
import { getShortcodesMock } from './getShortcodes';
import { getShortcodeMock } from './getShortcode';
import { getUsageMock } from './getUsage';
import { addContentProviderMock } from './addContentProvider';
import { addServiceRequestMock } from './addServiceRequest';
import { modifyShortcodeMock } from './modifyShortcode';
import { modifyContentProviderMock } from './modifyContentProvider';
import { deleteContentProviderMock } from './deleteContentProvider';

const handlers = [
  addContentProviderMock,
  addServiceRequestMock,
  getAdminOverviewMock,
  getCustomerOverviewMock,
  getContentProvidersMock,
  getServiceRequestsMock,
  getShortcodesMock,
  getShortcodeMock,
  getUsageMock,
  modifyShortcodeMock,
  modifyContentProviderMock,
  deleteContentProviderMock,
];

export default handlers;
