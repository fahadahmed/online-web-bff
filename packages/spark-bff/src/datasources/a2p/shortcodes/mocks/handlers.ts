import { getAdminShortcodesMock } from './getShortcodes';
import { getAdminShortcodesOverviewMock } from './getShortcodesOverview';

const handlers = [getAdminShortcodesMock, getAdminShortcodesOverviewMock];

export default handlers;
