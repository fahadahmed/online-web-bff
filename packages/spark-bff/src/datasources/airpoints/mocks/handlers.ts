import { rest } from 'msw';
import airpointsResponseMock from './airpoints-response.mock.json';
import addAirpointsResponseMock from './add-airpoints-response.mock.json';
import deleteAirpointsResponseMock from './delete-airpoints-response.mock.json';
import airpointsInactiveResponseMock from './airpoints-inactive-response.mock.json';
import addAirpointsErrorResponseMock from './add-airpoints-error-response.mock.json';
import deleteAirpointsErrorResponseMock from './delete-airpoints-error-response.mock.json';

interface AddAirpointsMockBody {
  accountNumber?: string;
  airpointsNumber?: string;
  firstName?: string;
  lastName?: string;
  status?: string;
}

export const airpointsResponseMocks = rest.get<AddAirpointsMockBody>(
  `http://testhost/v1/services/airpoints/:accountNumber`,
  async ({ params: { accountNumber } }, res, ctx) => {
    if (accountNumber === '123456789') {
      /* Simulate error responses */
      return res(ctx.status(200), ctx.json(airpointsResponseMock));
    }
    return res(ctx.status(200), ctx.json(airpointsInactiveResponseMock));
  },
);

export const updateAirpointsResponseMocks = rest.post<AddAirpointsMockBody>(
  `http://testhost/v1/services/airpoints/:accountNumber`,
  async (
    { params: { accountNumber }, body: { airpointsNumber } },
    res,
    ctx,
  ) => {
    /* add airpoints */
    if (airpointsNumber) {
      if (airpointsNumber === '911') {
        /* Simulate error responses */
        return res(ctx.status(400), ctx.json(addAirpointsErrorResponseMock));
      }
      return res(ctx.status(200), ctx.json(addAirpointsResponseMock));
    }
    /* delete airpoints */
    if (accountNumber) {
      if (accountNumber === '911') {
        /* Simulate error responses */
        return res(ctx.status(200), ctx.json(deleteAirpointsErrorResponseMock));
      }
      return res(ctx.status(200), ctx.json(deleteAirpointsResponseMock));
    }
    return null;
  },
);

const handlers = [airpointsResponseMocks, updateAirpointsResponseMocks];

export default handlers;
