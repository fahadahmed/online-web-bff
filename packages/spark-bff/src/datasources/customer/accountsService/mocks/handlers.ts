import { rest } from 'msw';
import accountSummaryMock from './account-summary.mock.json';
import emptyAccountListMock from './empty-account-list.mock.json';
import singleAccountListMock from './single-account-list.mock.json';
import accountListMock from './account-list.mock.json';
import guestAccountListMock from './guest-account-list.mock.json';
import unauthorizedAccountListMock from './unauthorized-account-list.mock.json';

const baseUrl = 'http://testhost/v1/customer/account';

export const accountSummaryHandler = rest.get<unknown, { accountId: string }>(
  `${baseUrl}/:accountId/summary`,
  async (req, res, ctx) => {
    const { accountId } = req.params;
    if (accountId.match(/^\d+$/) !== null) {
      return res(ctx.status(200), ctx.json(accountSummaryMock));
    }
    /* If the accountNumber does not match the given pattern, [0-9]+
    for example, 713012977, then it returns failure response, "Bad Request" */
    return res(ctx.status(400), ctx.json({}));
  },
);

export const accountListHandler = rest.get(
  `${baseUrl}/me/list`,
  async (_, res, ctx) => {
    return res(ctx.status(200), ctx.json(accountListMock));
  },
);

export const singleAccountListHandler = rest.get(
  `${baseUrl}/me/list`,
  async (_, res, ctx) => {
    return res(ctx.status(200), ctx.json(singleAccountListMock));
  },
);

export const emptyAccountListHandler = rest.get(
  `${baseUrl}/me/list`,
  async (_, res, ctx) => {
    return res(ctx.status(200), ctx.json(emptyAccountListMock));
  },
);

export const unauthorizedAccountListHandler = rest.get(
  `${baseUrl}/me/list`,
  async (_, res, ctx) => {
    return res(ctx.status(401), ctx.json(unauthorizedAccountListMock));
  },
);

export const guestAccountListHandler = rest.get(
  `${baseUrl}/me/list`,
  async (_, res, ctx) => {
    return res(ctx.status(403), ctx.json(guestAccountListMock));
  },
);

const handlers = [accountSummaryHandler, accountListHandler];

export default handlers;
