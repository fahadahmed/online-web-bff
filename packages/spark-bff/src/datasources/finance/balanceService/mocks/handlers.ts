import { rest } from 'msw';
import accountBalanceDetails from './account-balance.mock.json';

const baseUrl = 'http://testhost/v1/finance/balance/';

export const accountBalanceMock = rest.get<unknown, { accountNumber: string }>(
  `${baseUrl}account/:accountNumber`,
  async ({ params: { accountNumber } }, res, ctx) => {
    if (accountNumber.match(/^\d+$/) !== null) {
      return res(ctx.status(200), ctx.json(accountBalanceDetails));
    }
    /* If the accountNumber does not match the given pattern, [0-9]+
    for example, 713012977, then it returns failure response, "Bad Request" */
    return res(ctx.status(400), ctx.json({}));
  },
);

const handlers = [accountBalanceMock];

export default handlers;
