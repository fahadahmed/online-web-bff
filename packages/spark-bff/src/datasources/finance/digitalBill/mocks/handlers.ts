import { rest } from 'msw';
import accountBillingSummary from './account-billing-summary.mock.json';
import accountTransactions from './account-transactions.mock.json';

const baseUrl = 'http://testhost/v2/finance';

export const accountBillingSummaryMock = rest.get(
  `${baseUrl}/:accountNumber/bills/summary`,
  async (_, res, ctx) => {
    return res(ctx.json(accountBillingSummary));
  },
);

export const accountTransactionsMock = rest.get(
  `${baseUrl}/:accountNumber/bills/transactions`,
  async (_, res, ctx) => {
    return res(ctx.json(accountTransactions));
  },
);

const handlers = [accountBillingSummaryMock, accountTransactionsMock];

export default handlers;
