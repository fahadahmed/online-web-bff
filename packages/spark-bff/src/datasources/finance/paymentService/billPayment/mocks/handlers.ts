import { rest } from 'msw';
import poliPayBillSuccess from './poliPayBill-success.mock.json';
import poliPayBillValidationFailure from './poliPayBill-failure.mock.json';
import creditCardPayBillSuccess from './creditCard-payBill-success.mock.json';
import creditCardPayBillValidationFailure from './creditCard-payBill-failure.mock.json';
import deleteAutoPayBillSuccess from './delete-autoPayBill-success.mock.json';
import deleteAutoPayBillFailure from './delete-autoPayBill-failure.mock.json';

const baseUrl = 'http://testhost/v1/finance/payment';

export const addPoliPayBillMock = rest.post<unknown, { accountNumber: string }>(
  `${baseUrl}/polipay/bill/:accountNumber`,
  async ({ params: { accountNumber } }, res, ctx) => {
    /* If the accountNumber does not match the given pattern, [0-9]+
    for example, 789046383,then it returns failure response, "Request validation failure."  */
    if ((accountNumber as string).match(/^\d+$/) === null) {
      return res(ctx.status(400), ctx.json(poliPayBillValidationFailure));
    }
    return res(ctx.status(200), ctx.json(poliPayBillSuccess));
  },
);

export const addCreditCardPayBillMock = rest.post<
  unknown,
  { accountNumber: string }
>(
  `${baseUrl}/creditcard/bill/:accountNumber`,
  async ({ params: { accountNumber } }, res, ctx) => {
    /* If the accountNumber does not match the given pattern, [0-9]+
    for example, 789046383,then it returns failure response, "Request validation failure."  */
    if (accountNumber.match(/^\d+$/) === null) {
      return res(ctx.status(400), ctx.json(creditCardPayBillValidationFailure));
    }
    return res(ctx.status(200), ctx.json(creditCardPayBillSuccess));
  },
);

export const deleteAutoPayBillMock = rest.delete<
  unknown,
  { accountNumber: string; paymentMethodId: string }
>(
  `${baseUrl}/bill/:accountNumber/paymentsetup/:paymentMethodId`,
  async ({ params: { accountNumber, paymentMethodId } }, res, ctx) => {
    /** If accountNumber does not match the given pattern,
     for example, 746738998.
     Or If paymentMethodId does not match the given pattern,
     for example,"1-S0123-23"
     it returns failure response, "Request Validation Failure".
     */
    if (
      accountNumber.match(/^\d+$/) === null ||
      paymentMethodId.match(/^[\dA-Z][\dA-Z-]+[\dA-Z]$/) === null
    ) {
      return res(ctx.status(400), ctx.json(deleteAutoPayBillFailure));
    }
    return res(ctx.status(200), ctx.json(deleteAutoPayBillSuccess));
  },
);

const handlers = [
  addCreditCardPayBillMock,
  addPoliPayBillMock,
  deleteAutoPayBillMock,
];

export default handlers;
