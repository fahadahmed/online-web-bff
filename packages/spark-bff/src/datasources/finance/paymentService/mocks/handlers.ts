import { rest } from 'msw';
import SuccessResponseMock from './register-card-success-response.json';
import FailedResponse from './register-card-failed-response.json';

const baseUrl = 'http://testhost/v1/finance/payment';

export const registerCardHandler = rest.post(
  `${baseUrl}/creditcard/:lineNumber`,
  async ({ params: { lineNumber } }, res, ctx) => {
    if (lineNumber === '0276404521') {
      return res(ctx.status(500), ctx.json(FailedResponse));
    }
    return res(ctx.status(200), ctx.json(SuccessResponseMock));
  },
);

const handlers = [registerCardHandler];

export default handlers;
