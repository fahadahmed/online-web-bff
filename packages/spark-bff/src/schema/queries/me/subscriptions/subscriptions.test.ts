import { createTestClient, gql } from 'test/apolloTestUtils';
import { server } from 'test/server';
import handlers, {
  subscriptionOfferWhenNoLineAccessHandler,
  unhandledErrorSubscriptionsHandler,
} from 'datasources/line/lineSubscriptionsService/mocks/handlers';

test('Returns summary of subscription component offers for a given SparkId. Does not include billing information or CTAs.', async () => {
  server.use(...handlers);

  const { query } = createTestClient();
  const res = await query({
    query: gql`
      query {
        me {
          subscriptions {
            offer {
              offerId
              name
              productInstanceId
            }
            bundleOffer {
              bundleOfferId
              name
              productInstanceId
            }
            assetStartDate
            accountNumber
            lineNumber
            status {
              type
              label
            }
          }
        }
      }
    `,
  });

  expect(res.errors).toBeUndefined();
  expect(res.data!.me.subscriptions).toMatchSnapshot();
});

test('Returns empty list when there is no line access.', async () => {
  server.use(...[subscriptionOfferWhenNoLineAccessHandler]);

  const { query } = createTestClient();
  const res = await query({
    query: gql`
      query {
        me {
          subscriptions {
            offer {
              offerId
              name
              productInstanceId
            }
            bundleOffer {
              bundleOfferId
              name
              productInstanceId
            }
            assetStartDate
            accountNumber
            lineNumber
            status {
              type
              label
            }
          }
        }
      }
    `,
  });

  expect(res.errors).toBeUndefined();
  expect(res.data!.me.subscriptions).toMatchSnapshot();
});

test('Throws error when unsupported error code is returned.', async () => {
  server.use(...[unhandledErrorSubscriptionsHandler]);
  const { query } = createTestClient();
  const res = await query({
    query: gql`
      query {
        me {
          subscriptions {
            offer {
              offerId
              name
              productInstanceId
            }
            bundleOffer {
              bundleOfferId
              name
              productInstanceId
            }
            assetStartDate
            accountNumber
            lineNumber
            status {
              type
              label
            }
          }
        }
      }
    `,
  });

  expect(res.errors).not.toBeUndefined();
});
