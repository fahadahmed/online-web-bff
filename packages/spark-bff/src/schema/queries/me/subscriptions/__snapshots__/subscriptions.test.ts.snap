// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Returns empty list when there is no line access. 1`] = `Array []`;

exports[`Returns summary of subscription component offers for a given SparkId. Does not include billing information or CTAs. 1`] = `
Array [
  Object {
    "accountNumber": "369121518",
    "assetStartDate": "2021-10-12T03:31:03.964Z",
    "bundleOffer": null,
    "lineNumber": "0278675179",
    "offer": Object {
      "name": "Spotify Premium",
      "offerId": "misc077770",
      "productInstanceId": "1-2HAZ77770",
    },
    "status": Object {
      "label": "Active",
      "type": "ACTIVE",
    },
  },
  Object {
    "accountNumber": "369121518",
    "assetStartDate": "2021-03-13T03:31:03.964Z",
    "bundleOffer": null,
    "lineNumber": "0278675179",
    "offer": Object {
      "name": "Netflix Standard",
      "offerId": "misc050385",
      "productInstanceId": "1-2HA22220",
    },
    "status": Object {
      "label": "Active",
      "type": "ACTIVE",
    },
  },
  Object {
    "accountNumber": "369121518",
    "assetStartDate": "2021-10-12T03:31:03.964Z",
    "bundleOffer": null,
    "lineNumber": "0278675179",
    "offer": Object {
      "name": "Spotify Premium",
      "offerId": "misc052451",
      "productInstanceId": "1-2HAZ55550",
    },
    "status": Object {
      "label": "Processing",
      "type": "PROCESSING_ORDER_ADD",
    },
  },
  Object {
    "accountNumber": "369121518",
    "assetStartDate": "2021-11-18T03:31:03.964Z",
    "bundleOffer": Object {
      "bundleOfferId": "misc010921",
      "name": "Neon & Broadband",
      "productInstanceId": "1-2HAZ11111",
    },
    "lineNumber": "0278675179",
    "offer": Object {
      "name": "Neon",
      "offerId": "neon",
      "productInstanceId": "1-2HAZ11110",
    },
    "status": Object {
      "label": "Updating",
      "type": "PROCESSING_ACTIVATION",
    },
  },
  Object {
    "accountNumber": "369121518",
    "assetStartDate": "2021-09-22T03:31:03.964Z",
    "bundleOffer": Object {
      "bundleOfferId": "misc236443",
      "name": "Spark Sport",
      "productInstanceId": "1-2HA44441",
    },
    "lineNumber": "0278675179",
    "offer": Object {
      "name": "Spark Sport",
      "offerId": "spark_sport",
      "productInstanceId": "1-2HA44440",
    },
    "status": Object {
      "label": "Ready to activate",
      "type": "NOT_ACTIVE",
    },
  },
  Object {
    "accountNumber": "369121518",
    "assetStartDate": "2021-09-22T03:31:03.964Z",
    "bundleOffer": Object {
      "bundleOfferId": "misc583865",
      "name": "Sky Sport",
      "productInstanceId": "1-2HA33331",
    },
    "lineNumber": "0278675179",
    "offer": Object {
      "name": "Sky Sport",
      "offerId": "sky_sport",
      "productInstanceId": "1-2HA33330",
    },
    "status": Object {
      "label": "cancelled",
      "type": "CANCELLED",
    },
  },
  Object {
    "accountNumber": "369121518",
    "assetStartDate": "2021-10-12T03:31:03.964Z",
    "bundleOffer": null,
    "lineNumber": "0278675179",
    "offer": Object {
      "name": "McAfee Security Standard",
      "offerId": "mcafee-active",
      "productInstanceId": "1-VX31188",
    },
    "status": Object {
      "label": "Active",
      "type": "ACTIVE",
    },
  },
  Object {
    "accountNumber": "369121518",
    "assetStartDate": "2021-10-12T03:31:03.964Z",
    "bundleOffer": null,
    "lineNumber": "0278675179",
    "offer": Object {
      "name": "McAfee Security Standard",
      "offerId": "mcafee-not-active",
      "productInstanceId": "1-VX31189",
    },
    "status": Object {
      "label": "Ready to activate",
      "type": "NOT_ACTIVE",
    },
  },
]
`;
